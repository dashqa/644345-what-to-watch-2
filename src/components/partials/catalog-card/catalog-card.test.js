import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";
import CatalogCard from './catalog-card';

const props = {
  movie: {
    id: 1,
    name: `Macbeth`,
    previewImage: `http://image.com/image.png`,
    previewVideoLink: `http://video.com/video.mp4`
  },
  videoRef: React.createRef(),
  onEnter: () => {},
  onLeave: () => {},
};

it(`Catalog card correctly renders after relaunch`, () => {
  const tree = renderer.create(<BrowserRouter><CatalogCard {...props}/></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
