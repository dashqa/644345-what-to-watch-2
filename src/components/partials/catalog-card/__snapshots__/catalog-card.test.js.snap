// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog card correctly renders after relaunch 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    className="small-movie-card__link"
    href="/films/1"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className=""
        controls={false}
        height="180"
        muted={true}
        poster="http://image.com/image.png"
        width="280"
      >
        <source
          src="http://video.com/video.mp4"
          type="video/mp4"
        />
      </video>
      <h3
        className="small-movie-card__title"
      >
        Macbeth
      </h3>
    </div>
  </a>
</article>
`;
